core/src/main/scala/com/nvidia/spark/rapids/tool/analysis/AppSQLPlanAnalyzer.scala

```diff
@@ -65,6 +70,113 @@ class AppSQLPlanAnalyzer(app: AppBase, appIndex: Int) extends AppAnalysisBase(ap
     }
   }
 
+  /**
+   * Update the potential problems in the app object. This is a common function that is used by
+   * Both Qual/Prof analysis.
+   * For Qual apps, the app.sqlIDtoProblematic won't be set because it is done later during the
+   * aggregation phase.
+   * @param sqlId the SQL ID being analyzed
+   * @param potentialProblems a set of strings that represent the potential problems found in the
+   *                          SQL plan.
+   */
+  private def updateAppPotentialProblems(sqlId: Long,
+      potentialProblems: mutable.AbstractSet[String]): Unit = {
+    // Append problematic issues to the global variable for that SqlID
+    val existingIssues = app.sqlIDtoProblematic.getOrElse(sqlId, mutable.LinkedHashSet[String]())
+    app.sqlIDtoProblematic(sqlId) = existingIssues ++ potentialProblems
```

Comment:
why are we storing this same information twice?  Here and then below in string formation. Both are indexed by the sqlId so can't we look it up in either?

Reply:
Apparently this was used only for the output.
Removed the redundant field "`SQLExecutionInfoClass.problematic`"